console/algorithms/src/ecdsa/tests.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019-2025 Provable Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+const ITERATIONS: usize = 100;
+
+fn test_ecdsa<H: Hash<Output = Vec<bool>, Input = bool>>(hasher: &H, rng: &mut TestRng) {
+    for i in 1..ITERATIONS {
+        let (signing_key, message, signature) = test_helpers::sample_ecdsa_signature(i, hasher, rng);
+
+        // Construct the verifying key.
+        let verifying_key = VerifyingKey::from(&signing_key);
+        let verifying_key_bytes = verifying_key.to_encoded_point(true).as_bytes().to_vec();
+        let recovered_verifying_key = ECDSASignature::verifying_key_from_bytes(&verifying_key_bytes).unwrap();
+        assert_eq!(verifying_key, recovered_verifying_key);
+
+        // Verify the signature.
+        assert!(signature.verify(&verifying_key, hasher, &message.to_bits_le()).is_ok());
+        assert_eq!(signature.recover_public_key(hasher, &message.to_bits_le()).unwrap(), verifying_key);
+    }
+}
+
+#[test]
+fn test_ecdsa_signature() {
+    let rng = &mut TestRng::default();
+
+    test_ecdsa(&Keccak224::default(), rng);
+    test_ecdsa(&Keccak256::default(), rng);
+    test_ecdsa(&Keccak384::default(), rng);
+    test_ecdsa(&Keccak512::default(), rng);
+    test_ecdsa(&Sha3_224::default(), rng);
+    test_ecdsa(&Sha3_256::default(), rng);
+    test_ecdsa(&Sha3_384::default(), rng);
+    test_ecdsa(&Sha3_512::default(), rng);
+}
+
+#[test]
+fn test_ecdsa_signature_vector_1() {
```

Comment:
Is there any canonical set of test vectors which we can use, for ECDSA and/or Ethereum's usage?